testing/thrift/APKBUILD

```diff
@@ -0,0 +1,237 @@
+subpackages="
+    lib${pkgname}c:c
+    lib${pkgname}c-dev:cdev
+    lib${pkgname}:cpp
+    lib${pkgname}-dev:cppdev
+    lua-${pkgname}:lua
+    perl-${pkgname}:perl
+"
+# XXX: @CF: 20190508: Python support is broken. It might be better to use the pip version for the time being
+#/usr/bin/python3 test/test_sslsocket.py
+#ERROR:thrift.transport.TSSLSocket:failed to initialize SSL
+#Traceback (most recent call last):
+#  File "/home/alpine/thrift/src/thrift-0.11.0/lib/py/build/lib.linux-x86_64-3.7/thrift/transport/TSSLSocket.py", line 281, in _do_open
+#    return self._wrap_socket(plain_sock)
+#  File "/home/alpine/thrift/src/thrift-0.11.0/lib/py/build/lib.linux-x86_64-3.7/thrift/transport/TSSLSocket.py", line 179, in _wrap_socket
+#    self.ssl_context.set_ciphers(self.ciphers)
+#ssl.SSLError: ('No cipher can be selected.',)
+#    py-${pkgname}:py
+#    go-${pkgname}:go
+# XXX: @CF: 20180715: nodejs is working fine here, but users should probably just stick with "npm install thrift"
+#    node-${pkgname}:nodejs
+#    java-${pkgname}:java
+#    ruby-${pkgname}:ruby
+#    rust-${pkgname}:rust
+
+source="
+    ${pkgname}-$pkgver.tar.gz::https://github.com/apache/${pkgname}/archive/$pkgver.tar.gz
+    lib-perl-makefile-am-prefix.patch
+    lib-cpp-makefile-am-libadd-zlib-libevent.patch
+    lib-cpp-test-TNonblockingSSLServerTest-cpp-signal-h.patch
+    lib-cpp-test-Makefile-am-TEST-ENVIRONMENT.patch
+    configure-ac-lua-${_luaver}.patch
+    lib-cpp-mutex-adaptive-errorcheck-recursive-init.patch
+    lib-c_glib-test-testdebugproto-musl-broken-neginf.patch
+    lib-c_glib-test-testtransportsslsocket-musl-different-string-for-unknown-host.patch
+    lib-py-src-ext-endian-h-use-system-endian-h.patch
+    test-py-ignore-poorly-written-python-triggers-eaddrinuse.patch
+    configure-ac-boost-so.patch
+    configure-ac-define-boost-test-dyn-link.patch
+"
+# XXX: @CF: 20190508: breaks build for 0.12.0
+#lib-lua-makefile-am-longnumber.patch
+
+options="!strip"
+
+builddir="$srcdir/${pkgname}-$pkgver"
+
+prepare() {
+    default_prepare
+    cd "$builddir"
+    sh bootstrap.sh
+}
+
+# XXX: @CF: 20180612: --with-java requires java and javac to be in the path #export PATH=${PATH}:/usr/lib/jvm/java-1.8-openjdk/bin
+# XXX: @CF: 20180612: ax_lua.m4 is broken (i.e. does not recognize LUA_LIB=$(pkg-config --libs lua${_luaver}) properly), so force lua into cppflags / ldflags
+build() {
+	cd "$builddir"
+	
+	LUA_INCLUDE="$(pkg-config --cflags lua${_luaver}) -DLUA_COMPAT_MODULE" \
+	LUA_LIB="$(pkg-config --libs lua${_luaver})" \
+	LDFLAGS="-Wl,--no-undefined" \
+	./configure \
+		--build=$CBUILD \
+		--host=$CHOST \
+		--prefix=/usr \
+		--sysconfdir=/etc \
+		--mandir=/usr/share/man \
+		--localstatedir=/var \
+		--without-nodejs \
+		--without-go \
+		--without-python \
+		--without-python3
+	make
+}
+
+check() {
+    cd "$builddir"
+    make check
+}
+
+package() {
+    local tpkgdir="${pkgdir}-tmp"
+    cd "$builddir"
+    make DESTDIR="$tpkgdir" install
```

Comment:
you can install to $pkgdir and then move from there, no need to do this dance around a mythical -tmp package dir